Crash when bookmark URL is not accessible

## Step to reproduce:

1. Start up `shiori serve` and login
2. Add a bookmark that is not accessible for shiori
    1. If the bookmark is likely i/o timeout, need to click the `Update archive` on the saved bookmark, the options `Keep the old title and excerpt` and `Update archive as well` is not necessary to check or uncheck.
    2. If the bookmark is likely `connection refused`, the shiori will crash immediately.
4. shiori crash

## Log is following:

```
2022/08/04 08:57:14 error downloading boorkmark: error downloading bookmark: Get "http://127.0.0.1:2323": dial tcp 127.0.0.1:2323: connect: connection refused
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xaaf64a]

goroutine 68 [running]:
github.com/go-shiori/shiori/internal/webserver.(*handler).apiInsertBookmark.func1()
        /home/runner/work/shiori/shiori/internal/webserver/handler-api.go:329 +0x8a
created by github.com/go-shiori/shiori/internal/webserver.(*handler).apiInsertBookmark
        /home/runner/work/shiori/shiori/internal/webserver/handler-api.go:324 +0x5c5
```

The results is same when the error message is `i/o timeout`, `read: connection reset by peer` or `connection refused` like above.

## Version

`shiori_1.5.3_Linux_x86_64.tar.gz` from GitHub release page.

I think it still exist, I'm on commit `d15dc186741c8b8c37949e8b0ef6e7e6273714e8`.

I build the shiori by run `go get -u -v github.com/go-shiori/shiori` and then `go build`, and get binary `shiori`.

After configuring database with MariaDB by environment variables like follows.

```
export SHIORI_DBMS=mysql
export SHIORI_MYSQL_USER=tst
export SHIORI_MYSQL_PASS=test
export SHIORI_MYSQL_NAME=test
export SHIORI_MYSQL_ADDRESS=tcp(127.0.0.1:3306)
```

Run `./shiori --portable migrate` and `./shiori --portable serve`.

Then login the web UI and type an inaccessible URL like `http://localhost:2323`, and I got the final crash log.

```
INFO[0053] POST /api/bookmarks                           proto=HTTP/1.1 remote="127.0.0.1:50046" reqlen=97 size=223 status=200
2022/10/09 22:18:54 error downloading boorkmark: error downloading bookmark: Get "http://localhost:2323": dial tcp [::1]:2323: connect: connection refused
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xaf85b7]

goroutine 374 [running]:
github.com/go-shiori/shiori/internal/webserver.(*handler).apiInsertBookmark.func1()
        /home/leafee98/Desktop/project/shiori/internal/webserver/handler-api.go:342 +0x97
created by github.com/go-shiori/shiori/internal/webserver.(*handler).apiInsertBookmark
        /home/leafee98/Desktop/project/shiori/internal/webserver/handler-api.go:337 +0x62e
```

If there are any wrong in my building steps or configuration or usage, let me know.

--- a/internal/webserver/handler-api.go
+++ b/internal/webserver/handler-api.go
@@ -26,7 +26,7 @@ import (
 func downloadBookmarkContent(book *model.Bookmark, dataDir string, request *http.Request) (*model.Bookmark, error) {
 	content, contentType, err := core.DownloadBookmark(book.URL)
 	if err != nil {
-		return nil, fmt.Errorf("error downloading bookmark: %s", err)
+		return nil, fmt.Errorf("error downloading url: %s", err)
 	}
 
 	processRequest := core.ProcessRequest{
@@ -40,7 +40,7 @@ func downloadBookmarkContent(book *model.Bookmark, dataDir string, request *http
 	content.Close()
 
 	if err != nil && isFatalErr {
-		panic(fmt.Errorf("failed to process bookmark: %v", err))
+		return nil, fmt.Errorf("failed to process: %v", err)
 	}
 
 	return &result, err
@@ -328,6 +328,7 @@ func (h *handler) apiInsertBookmark(w http.ResponseWriter, r *http.Request, ps h
 			bookmark, err := downloadBookmarkContent(book, h.DataDir, r)
 			if err != nil {
 				log.Printf("error downloading boorkmark: %s", err)
+				return
 			}
 			if _, err := h.DB.SaveBookmarks(context.Background(), false, *bookmark); err != nil {
 				log.Printf("failed to save bookmark: %s", err)
@@ -339,8 +340,7 @@ func (h *handler) apiInsertBookmark(w http.ResponseWriter, r *http.Request, ps h
 		book, err = downloadBookmarkContent(book, h.DataDir, r)
 		if err != nil {
 			log.Printf("error downloading boorkmark: %s", err)
-		}
-		if _, err := h.DB.SaveBookmarks(ctx, false, *book); err != nil {
+		} else if _, err := h.DB.SaveBookmarks(ctx, false, *book); err != nil {
 			log.Printf("failed to save bookmark: %s", err)
 		}
 	}

